### ISSUE TITLE
staticcheck: detect incorrect declaration of typed enums

### ISSUE THREAD

**Issue by dominikh** [dominikh/go-tools#220]
A common beginner mistake is writing code like this:

```
const (
	EnumFoo SomeType = 1
	EnumBar          = 2
)
```

expecting EnumBar to have type SomeType.

**Comment by dominikh** [comment]
Real world example from https://github.com/golang/go/blob/1125fae989a3016d509c23fee15793b231e5e8e1/src/bytes/buffer.go#L34

```
const (
	opRead      readOp = -1 // Any other read operation.
	opInvalid          = 0  // Non-read operation.
	opReadRune1        = 1  // Read rune of size 1.
	opReadRune2        = 2  // Read rune of size 2.
	opReadRune3        = 3  // Read rune of size 3.
	opReadRune4        = 4  // Read rune of size 4.
)
```

The code used to read

```
 const (
       opInvalid  readOp = iota // Non-read operation.
       opReadRune               // Read rune.
       opRead                   // Any other read operation.
 )
```

and was incorrectly modified to its current form.

**Comment by martisch** [comment]
Thanks for the notification. Will have a look at correcting. Note that the iota was removed by request during review but i indeed missed this would change the type.
https://go-review.googlesource.com/#/c/go/+/28817/2..4/src/bytes/buffer.go

**Comment by dominikh** [comment]
Implemented in fade51a
